packages/animations/test/page_transition_switcher_test.dart

```diff
@@ -0,0 +1,622 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/animations.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('transitions in a new child.', (WidgetTester tester) async {
+    final UniqueKey containerOne = UniqueKey();
+    final UniqueKey containerTwo = UniqueKey();
+    final UniqueKey containerThree = UniqueKey();
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: Container(key: containerOne, color: const Color(0x00000000)),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    Map<Key, double> _primaryAnimation =
+        _getPrimaryAnimation(<Key>[containerOne], tester);
+    Map<Key, double> _secondaryAnimation =
+        _getSecondaryAnimation(<Key>[containerOne], tester);
+    expect(_primaryAnimation[containerOne], moreOrLessEquals(1.0));
+    expect(_secondaryAnimation[containerOne], moreOrLessEquals(0.0));
+
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: Container(key: containerTwo, color: const Color(0xff000000)),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+    await tester.pump(const Duration(milliseconds: 40));
+
+    _primaryAnimation =
+        _getPrimaryAnimation(<Key>[containerOne, containerTwo], tester);
+    _secondaryAnimation =
+        _getSecondaryAnimation(<Key>[containerOne, containerTwo], tester);
+    // Secondary is running for outgoing widget.
+    expect(_primaryAnimation[containerOne], moreOrLessEquals(1.0));
```

Comment:
Maybe moreOrLessEquals isn't needed for cases were we believe the animation's value will have reached 0 or 1